tests/test_cli_compile.py

```diff
@@ -1035,8 +1034,11 @@ def test_sub_dependencies_with_constraints(pip_conf, runner):
 
     req_out_lines = set(out.stderr.splitlines())
     assert {
-        "small-fake-a==0.1         # via small-fake-with-unpinned-deps",
-        "small-fake-b==0.2         # via small-fake-with-unpinned-deps",
-        "small-fake-with-deps-and-sub-deps==0.1",
-        "small-fake-with-unpinned-deps==0.1  # via small-fake-with-deps-and-sub-deps",
+        "small-fake-a==0.1         "
+        "# via -c constraints.txt (line 1), small-fake-with-unpinned-deps",
+        "small-fake-b==0.2         "
+        "# via -c constraints.txt (line 2), small-fake-with-unpinned-deps",
+        "small-fake-with-deps-and-sub-deps==0.1  # via -r requirements.in (line 2)",
+        "small-fake-with-unpinned-deps==0.1  "
+        "# via -c constraints.txt (line 3), small-fake-with-deps-and-sub-deps",
```

Comment:
I believe we could use heere `--no-annotation` also and add a separate test test for annotations (or adjust test_annotate_option).

Reply:
:man_shrugging: Sure, done in the last commit, which I'll leave unsquashed for now for simpler review.

Reply:
Thanks!